cpp/include/kvikio/shim/cufile.hpp

```diff
@@ -74,6 +83,26 @@ class cuFileAPI {
     get_symbol(DriverSetMaxCacheSize, lib, KVIKIO_STRINGIFY(cuFileDriverSetMaxCacheSize));
     get_symbol(DriverSetMaxPinnedMemSize, lib, KVIKIO_STRINGIFY(cuFileDriverSetMaxPinnedMemSize));
 
+#ifdef CUFILE_BATCH_API_FOUND
+    get_symbol(BatchIOSetUp, lib, KVIKIO_STRINGIFY(cuFileBatchIOSetUp));
+    get_symbol(BatchIOSubmit, lib, KVIKIO_STRINGIFY(cuFileBatchIOSubmit));
+    get_symbol(BatchIOGetStatus, lib, KVIKIO_STRINGIFY(cuFileBatchIOGetStatus));
+    get_symbol(BatchIOCancel, lib, KVIKIO_STRINGIFY(cuFileBatchIOCancel));
+    get_symbol(BatchIODestroy, lib, KVIKIO_STRINGIFY(cuFileBatchIODestroy));
+
+    // HACK: we use the mangled name of the `cuFileBatchContextState` to determine if cuFile's
+    // batch API is available. Notice, the symbols of `cuFileBatchIOSetUp` & co. exist all
+    // the way back to CUDA v11.5 but calling them is undefined behavior.
+    // TODO: when CUDA v12.2 is released, use `cuFileReadAsync` to determine the availability of
+    // both the batch and async API.
+    try {
+      void* s{};
+      get_symbol(s, lib, "_ZTS23cuFileBatchContextState");
```

Comment:
No, not AFAIK :/
I think this will suffice for now. When CUDA v12.2 is released, I think we should make the Batch API require v12.2 and then use `cuFileReadAsync` to determine the availability of both the batch and async API.